### PR TITLE
[Contrib] Logrotate

### PR DESCRIPTION
**Pull request by bagasme**
This PR adds sample `logrotate` configuration and corresponding docs entry.

### CODE REVIEW

**Comment by zeripath** [comment]
You know there is built in log rotation? It's slightly different from logrotate but it is there.

**Comment by bagasme** [comment]
Don't merge until #9869 feature is available, to let users choose between Gitea's implementation or external tools.

**Comment by zeripath** [comment]
Please see: https://docs.gitea.io/en-us/config-cheat-sheet/#file-log-mode-log-file-log-file-or-mode-file

**Comment by bagasme** [comment]
@zeripath Aha! I overlooked it.

**Comment by techknowlogick** [comment]
I'm unsure of merging this PR as-is, as it is pretty generic and other guides on configuring logrotate might be better. Especially since it would need customization based on the specific user setup (some downstream linux distributions have different settings than what is defined on the "install from source page"), and due to that I suspect this may add to confusion about configuration.

Maybe a quick "Gitea includes log rotation built-in, but if you want to use logrotate here is the setting you need to disable" (or something along those lines) and you can put it on the logging page: https://docs.gitea.io/en-us/logging-configuration/

**Comment by bagasme** [comment]
@techknowlogick For your suggestion I will address on another PR.

File: contrib/logrotate/gitea.conf [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+/var/lib/gitea/log/*.log, /var/lib/gitea/log/hooks/*.log {
+    daily
+    missingok
+    rotate 7
+    compress
+    notifempty
+    create 0660 git git
+    sharedscripts
+    postrotate
+        systemctl restart gitea.service
```
Comment (silverwind) [inline contrib/logrotate/gitea.conf:13]: This restart is unnecessary, it's probably enough to enable `copytruncate` and remove both the `postrotate` and `create` options.

Comment (bagasme) [reply contrib/logrotate/gitea.conf:13]: No no no,

Many application use restart-after-logrotate technique, and Gitea should too.

Comment (silverwind) [reply contrib/logrotate/gitea.conf:13]: That is a recipe for disaster. What if the application fails to restart? You certainly don't want logrotate to be able to cause a service outage.

Comment (bagasme) [reply contrib/logrotate/gitea.conf:13]: @silverwind hmm... `logrotate` should be blamed

**Comment by techknowlogick** [comment]
Closing per 9930

**Pull request closed without merging**